gigascience/gigadb-website
# Sort dataset author options

# Pull request for issue: #1768

* Sort the author options in the adminDatasetAuthor form, e.g http://gigadb.gigasciencejournal.com/adminDatasetAuthor/create

## How to test?

* Navigate to http://gigadb.gigasciencejournal.com/adminDatasetAuthor/create
* Open author dropdown and confirm it is sorted alphabetically

## How have functionalities been implemented?

* Updated custom component `DropdownField`
  * add `enableSorting` option which is false by default
  * Sort by displayed text in the component:

```php
        if ($this->enableSorting) {
          // expects $data elements to have $textField key with string value, might need to be extended for more use cases
          usort($data, function ($a, $b) use ($textField) {
            return strcmp($a[$textField], $b[$textField]);
          });
        }
```

## Any issues with implementation?

--

## Any changes to automated tests?

--
